src/views/ReferralProgram.vue

```diff
@@ -254,13 +357,18 @@ export default class ReferralProgram extends Mixins(
       });
     }
   }
+
+  async handleSetReferrer(event: Event): Promise<void> {
```

Comment:
You can use `.stop` event modificator in template to stop propagation:
`@click.stop="handleSetReferrer"`